gem5-configs/configs-boot-tests/system/ruby_system.py

```diff
@@ -77,6 +77,9 @@ def __init__(self, kernel, disk, cpu_type, mem_sys, num_cpus, opts):
         elif mem_sys == 'MESI_Two_Level':
             from MESI_Two_Level import MESITwoLevelCache
             self.caches = MESITwoLevelCache()
+        elif mem_sys == 'MOESI_CMP_directory':
```

Comment:
I'm going to leave the below for posterity, but it's not going to work. The problem is that each of the modules (e.g., `MIExampleSystem`, `MESITwoLevel`, etc.) can only be imported if we've built for that protocol. We can't import more than one at any time. This is one of the key problems we'll need to solve to make these scripts more flexible. 

We want to be able to do something like the below, but we can't right now.

```
from MI_example_caches import MIExampleSystem
from MESI_Two_Level import MESITwoLevelCache
from MOESI_CMP_directory import MOESICMPDirCache
ruby_protocols = {
    'MI_example': MIExampleSystem,
    'MESI_Two_Level': MESITwoLevelCache,
    'MOESI_CMP_directory': MOESICMPDirCache
}

self.caches = ruby_protocols[mem_sys]()
```